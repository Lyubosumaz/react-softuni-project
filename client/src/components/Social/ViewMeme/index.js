import { useState, useEffect } from 'react';
import { PropTypes } from 'prop-types';
import { httpSocial } from '../../../services/http';
import MemeCard from '../../MemeCard';
import { componentData } from '../../../utils/class-names.json';

export default function ViewMeme({ memeId }) {
    const memeIdProps = memeId;
    const [meme, setMeme] = useState(null);

    useEffect(() => {
        httpSocial.getMeme(memeIdProps).then((meme) => {
            setMeme(meme);
        });
    }, [memeIdProps]);

    return <section className={`${componentData}`}>{meme && <MemeCard num={0} meme={meme} />}</section>;
}

ViewMeme.propTypes = {
    memeId: PropTypes.string.isRequired,
};
