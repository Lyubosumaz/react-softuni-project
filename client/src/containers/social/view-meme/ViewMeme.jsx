import React, { useState, useEffect } from 'react';
import MemeCard from '../components/meme-card/MemeCard';
import http from '../../../services/http';
import handleRoute from '../../../utils/handleRoutes';

export default function ViewMeme(props) {
    const memeId = props.match.params.id;
    const [meme, setMeme] = useState(null)

    useEffect(() => {
        http.Social.getMeme(memeId)
            .then((meme) => { setMeme(meme) })
    }, []);

    return (
        <div className="main-container">
            <h1>View Meme</h1>

            {meme && <MemeCard meme={meme} />}


            <div className="info-container">
                <p>Add Meme or just watch others memes <button className="info-button" 
                onClick={handleRoute('/social')}>Back</button>.</p>
            </div>
        </div>
    );
};
