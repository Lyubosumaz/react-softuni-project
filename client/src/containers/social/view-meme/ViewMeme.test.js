import React from 'react';
import { shallow } from 'enzyme';
import ViewMeme from './ViewMeme';

it('Renter HouseOfFame component', () => {
    const wrapper = shallow(<ViewMeme match={{ params: { id: 'test' } }} />);
    expect(wrapper).toBeTruthy();
});

it('Checks for html changes', () => {
    const wrapper = shallow(<ViewMeme match={{ params: { id: 'test' } }} />);
    expect(wrapper).toMatchSnapshot();
});
