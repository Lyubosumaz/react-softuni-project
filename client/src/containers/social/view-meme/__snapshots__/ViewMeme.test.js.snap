// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks for html changes 1`] = `
<div
  className="main-container"
>
  <h1>
    View Meme
  </h1>
  <div
    className="info-container"
  >
    <p>
      Add Meme or just watch others memes 
      <button
        className="info-button"
        onClick={[Function]}
      >
        Back
      </button>
      .
    </p>
  </div>
</div>
`;
